mobile/v1/bigquery/sql-runner/sql/standard/02-screen-views/01-main/00-setup-screen-views.sql

```diff
@@ -0,0 +1,225 @@
+  run_id TIMESTAMP,
+  model_version STRING,
+  model STRING,
+  module STRING,
+  run_start_tstamp TIMESTAMP,
+  run_end_tstamp TIMESTAMP,
+  rows_this_run INT64,
+  distinct_key STRING,
+  distinct_key_count INT64,
+  time_key STRING,
+  min_time_key TIMESTAMP,
+  max_time_key TIMESTAMP,
+  duplicate_rows_removed INT64,
+  distinct_keys_removed INT64
+)
+PARTITION BY DATE(run_start_tstamp);
+
+-- Setup temp metadata tables for this run
+CREATE OR REPLACE TABLE {{.scratch_schema}}.mobile_sv_metadata_this_run{{.entropy}} (
+  id STRING,
+  run_id TIMESTAMP,
+  model_version STRING,
+  model STRING,
+  module STRING,
+  run_start_tstamp TIMESTAMP,
+  run_end_tstamp TIMESTAMP,
+  rows_this_run INT64,
+  distinct_key STRING,
+  distinct_key_count INT64,
+  time_key STRING,
+  min_time_key TIMESTAMP,
+  max_time_key TIMESTAMP,
+  duplicate_rows_removed INT64,
+  distinct_keys_removed INT64
+);
+
+INSERT INTO {{.scratch_schema}}.mobile_sv_metadata_this_run{{.entropy}} (
+  SELECT
+    'run',
+    run_id,
+    '{{.model_version}}',
+    'mobile',
+    'screen-views',
+    CURRENT_TIMESTAMP(),
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL
+
+  FROM {{.scratch_schema}}.mobile_metadata_run_id{{.entropy}}
+);
+
+-- Create page views table if it doesn't exist
+CREATE TABLE IF NOT EXISTS {{.output_schema}}.mobile_screen_views{{.entropy}} (
+
+  screen_view_id STRING NOT NULL,
+  event_id STRING,
+  app_id STRING,
+  user_id STRING,
+  device_user_id STRING,
+  network_userid STRING,
+  session_id STRING,
+  session_index INT64,
+  previous_session_id STRING,
+  session_first_event_id STRING,
+  screen_view_in_session_index INT64,
+  screen_views_in_session INT64,
+  dvce_created_tstamp TIMESTAMP,
+  collector_tstamp TIMESTAMP,
+  derived_tstamp TIMESTAMP,
+  screen_view_name STRING,
+  screen_view_transition_type STRING,
+  screen_view_type STRING,
+  screen_fragment STRING,
+  screen_top_view_controller STRING,
+  screen_view_controller STRING,
+  screen_view_previous_id STRING,
+  screen_view_previous_name STRING,
+  screen_view_previous_type STRING,
+  platform STRING,
+  dvce_screenwidth INT64,
+  dvce_screenheight INT64,
+  device_manufacturer STRING,
+  device_model STRING,
+  os_type STRING,
+  os_version STRING,
+  android_idfa STRING,
+  apple_idfa STRING,
+  apple_idfv STRING,
+  open_idfa STRING,
+  device_latitude FLOAT64,
+  device_longitude FLOAT64,
+  device_latitude_longitude_accuracy FLOAT64,
+  device_altitude FLOAT64,
+  device_altitude_accuracy FLOAT64,
+  device_bearing FLOAT64,
+  device_speed FLOAT64,
+  geo_country STRING,
+  geo_region STRING,
+  geo_city STRING,
+  geo_zipcode STRING,
+  geo_latitude FLOAT64,
+  geo_longitude FLOAT64,
+  geo_region_name STRING,
+  geo_timezone STRING,
+  user_ipaddress STRING,
+  useragent STRING,
+  carrier STRING,
+  network_technology STRING,
+  network_type STRING,
+  build STRING,
+  version STRING,
+  load_tstamp TIMESTAMP
```

Comment:
Nit: Something strikes me as off about the name `load_tstamp` - feels like it should be the time an event was loaded to database. Perhaps `model_tstamp` or `datamodel_tstamp`, or some other name might fit?

Reply:
Sure. I am easy on this one. Will change